pkg/deprecatetemplates/enforcer.go

```diff
@@ -0,0 +1,81 @@
+package deprecatetemplates
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowplugins "k8s.io/test-infra/prow/plugins"
+
+	"github.com/openshift/ci-tools/pkg/rehearse"
+)
+
+type Enforcer struct {
+	Enabled bool
+
+	existingTemplates sets.String
+	allowlist         *Allowlist
+}
+
+func (e *Enforcer) loadExistingTemplates(cuCfg prowplugins.ConfigUpdater) {
+	templates := sets.NewString()
+	templatePathPrefix := "ci-operator/templates"
+	for pattern, cmSpec := range cuCfg.Maps {
+		if strings.HasPrefix(pattern, templatePathPrefix) {
+			templates.Insert(cmSpec.Name)
+		}
+	}
+
+	e.existingTemplates = templates
+}
+
+func (e *Enforcer) ingest(job prowconfig.JobBase) {
+	for template := range e.existingTemplates {
+		if rehearse.UsesConfigMap(job, template) {
+			e.allowlist.Insert(job.Name, template)
```

Comment:
Shouldn't the `allowList.Insert` be gated by an `if update` check? Right now the verification can never fail